[STORM-2071]  Add in retry after rebalance in unit test

ppoulosk opened the pull request:
This retries the command a five times after rebalance, as it was failing on some systems.

abellina commented on storm-core/test/clj/org/apache/storm/nimbus_test.clj:
```diff
@@ -1218,8 +1218,16 @@
+              ;; One second doesn't work on all systems.  Try it five times
+              (loop [n 5]
+                (if-let [result (try
+                                  [(Thread/sleep 1000)
```
how about:
(try
  (do
     (Thread/sleep 1000)
     (.deactivate nimbus "t1"))
  (catch ...

HeartSaVioR replied:
+1 to this one. Seems clearer.

ppoulosk replied:
Agreed.  Will fix.  Thanks, @abellina.

abellina commented on storm-core/test/clj/org/apache/storm/nimbus_test.clj:
```diff
@@ -1218,8 +1218,16 @@
+                                  (catch Exception e
+                                    (when (zero? n)
+                                      (throw e))))]
+                         (result 0)
```
these should be aligned with (if-let +2 spaces from the opening parenthesis.

HeartSaVioR commented:
+1 once comments from @abellina are addressed. Nice finding.

ppoulosk commented:
Coded fix, retested, and pushed.  Thanks!!

abellina commented:
+1 thanks @ppoulosk

HeartSaVioR commented:
From JIRA comment @revans2 said he fixed this issue from Supervisor V2 patch.

@revans2 Since we saw builds failing (high chance from my dev.) how about addressing build issue before than Supervisor V2?

@ppoulosk @revans2 
Please have a cross look at each patch. We need to choose one which will be merged.

revans2 commented:
I would be happy to pull that fix back into a separate patch.  @ppoulosk this patch looks fine to me, my only concern would be documenting it a little more explaining that the state change might not have completed yet, and until it does we cannot rebalance.  Alternatively if you want to pull in my changes, by replacing sleep with the following I would be OK with that too.

```
(defn wait-for-status [nimbus name status]
  (while-timeout 5000
    (let [topo-summary (first (filter (fn [topo] (= name (.get_name topo))) (.get_topologies (.getClusterInfo nimbus))))
          topo-status (if topo-summary (.get_status topo-summary) "NOT-RUNNING")]
      (log-message "WAITING FOR "name" TO BE " status " CURRENT " topo-status)
      (not= topo-status status))
    (Thread/sleep 100)))
...
(wait-for-status nimbus "t1" "ACTIVE") 
```

Either solution is fine with me.

HeartSaVioR commented:
@ppoulosk Do you have any update on this?

ppoulosk commented:
@HeartSaVioR @revans2 I have no preference which solution gets pulled in first.  

I fixed the comment, retested, squashed the three commits into one and pushed it to my branch.

Thanks!

HeartSaVioR commented:
@ppoulosk Could we close this as Supervisor V2 was merged? Thanks.

ppoulosk commented:
Yes.  Thanks!
